### ISSUE TITLE
[a11y] QExpansionItem - use of 'aria-expanded' and 'aria-label'

### ISSUE THREAD

**Issue by Evertvdw** [quasarframework/quasar#14338]
### What happened?

The expansion item currently does not provide an `aria-expanded` indicating the state of the item. Also we should be able to add an `aria-label` attribute to everything with a tabindex of 0  within that component so we can signal to screen readers what will happen when using that button/link.

### What did you expect to happen?

Expansion item should be accessible

### Reproduction URL

https://codepen.io/Mett-codepens/pen/wvjKjpx

### How to reproduce?

Use the element inspector inside the codepen to see the issue. The element that toggles the expansion item should have an `aria-expanded` property.

### Flavour

UMD

### Areas

Components (quasar)

### Platforms/Browsers

Firefox, Chrome, Safari, Microsoft Edge, iOS, Android, Electron

**Comment by pdanpdan** [comment]
About `aria-expanded` I don't know how we missed it :(
About `aria-label` - it's a little bit more complicated - have to think about it
Thanks for reporting

**Comment by pdanpdan** [comment]
Hello. I was working on some changes for a11y and I need some help with testing.

The changes are related to:
- feat(QExpansionItem): add a11y for toggle buttons; improve header slot scope quasarframework#14338
- feat(QSelect): improve a11y - aria-readonly on focusable element; remove aria-activedescendant duplicate
- feat(QChip): add aria-label and proper role for remove button/icon
- feat(QDialog): set aria-modal on dialog
- feat(QMenu): set default role of menu
- feat(QFab): add a11y for toggle button; describe aria-controls/own
- feat(QBtnDropdown): add a11y for toggle buttons; describe aria-controls/own

I have published a test documentation site at https://pdanpdan.github.io/quasar-docs/ using the changed code.
If you use screen readers and/or a11y checking tools can you please check the pages for the changed components and report if there are problems or if there are improvements?
From the docs you can open codepens with the changed code, so if you have special cases you want to test you can.

**Comment by Evertvdw** [comment]
@pdanpdan Thanks for picking this up!  I checked the QExpansionItem and it seems to work correctly.

One thing I think is missing still is a default `aria-label` for the expand button. I think Quasar should provide it's own label when none is supplied so that the component is as accessible as possible when using it out of the box. My suggestion would be to put the string `Expand...` / `Collapse...` inside there where you fill in the supplied label of the component.

**Comment by pdanpdan** [comment]
I added `toggle-aria-label` prop for the label. I'm not really sure a default label (Expand/Collapse) would have meaning, because it would miss the "what it expands/collapses".
I'll wait and see for @rstoenescu opinion.

I mean orca/gnome/chrome/ubuntu already announces expanded/collapsed when the element gets the focus

**Comment by Evertvdw** [comment]
@pdanpdan What I mean is that you can prepend `expand/collapse` to the `label` property if that is supplied to use that as the aria-label by default.

**Comment by pdanpdan** [comment]
Anyone that is native in different languages except English please take a look at https://github.com/quasarframework/quasar/pull/14388 and check what funny things I wrote there in the translations (and comment there with the right variant)

**Comment by rstoenescu** [comment]
Enhancements will be available in Quasar v2.8.4 and 1.20.2
